[jktang14] iP

jktang14 opened the pull request:
# Azir
> "Great things are done by a series of small things brought together." – Vincent Van Gogh [Source](https://motiva.art/blog/vincent-van-gogh-quotes-inspire-artistic-journey/#:~:text=%E2%80%9CGreat%20things%20are%20done%20by,of%20small%20things%20brought%20together.%E2%80%9D)

Azir frees your mind from having to remember what tasks to do
- Easy to learn
- ~~FAST~~ ***SUPER FAST*** to use

General steps to communicate with the chatbot
1. Run the chatbot
2. Input either todo, deadline or event command
3. Type bye to exit the chatbot

Features:
- [x] Managing tasks
- [ ] Reminders (Coming soon)

Here is Java's `main` code
```Java
public static void main(String[] args) {
    System.out.println("Hello World")
}
```

This is all **FREE!** Enjoy 👍

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jktang14 committed:
Remove duke.java

jktang14 committed:
Add Azir.java

jktang14 committed:
Add greeting message to bot

jktang14 committed:
Add echo functionality

jktang14 committed:
Add List.java

jktang14 committed:
Remove List.java

jktang14 committed:
Add support for list

jktang14 committed:
Add Task.java

jktang14 committed:
Add String representation for Task

jktang14 committed:
Add methods for setting Task status

jktang14 committed:
Add Task functionality to chatbot

jktang14 committed:
Add Todo.java

jktang14 committed:
Add Deadline.java

jktang14 committed:
Add Event.java

jktang14 committed:
Edit constructor of Task

jktang14 committed:
Add string representation for Todo

jktang14 committed:
Add string representation for Deadline

jktang14 committed:
Add string representation for Event

jktang14 committed:
Add support for 3 tasks

jktang14 committed:
Add task print statements

jktang14 committed:
Edit message for list of tasks

jktang14 committed:
Edit file path of chatbot

jktang14 committed:
Add input to input.txt

jktang14 committed:
Add expected output

jktang14 committed:
Add AzirException.java

jktang14 committed:
Add constructor for AzirException

jktang14 committed:
Catch error for invalid input

jktang14 committed:
Handle error for empty todo description

jktang14 committed:
Handle exceptions for deadline input

jktang14 committed:
Handle exceptions for event

jktang14 committed:
Handle exception for marking and unmarking

jktang14 committed:
Add task deletion logic

jktang14 committed:
Handle writes to file

jktang14 committed:
Remove print statement

jktang14 committed:
Add loading task functionality

jktang14 committed:
Add formatting for hard disk file

jktang14 committed:
Modify Azir for new read and write format

jktang14 committed:
Merge branch 'branch-Level-7'

jktang14 committed:
Modify expected output

jktang14 committed:
Edit date input of deadline task

jktang14 committed:
Edit String formatting of deadline task

jktang14 committed:
Add error message for improper date format

jktang14 committed:
Merge branch 'branch-Level-8'

jktang14 committed:
Add TaskList

jktang14 committed:
Add TaskList manipulation methods

jktang14 committed:
Add Storage

jktang14 committed:
Add read and write to Storage

jktang14 committed:
Add Parser

jktang14 committed:
Add parsing logic to Parser

jktang14 committed:
Add Ui

jktang14 committed:
Add Ui messages

jktang14 committed:
Add method to get list size

jktang14 committed:
Add parsing for bye command

jktang14 committed:
Add function to get task

jktang14 committed:
Edit deadline inputs

jktang14 committed:
Edit expected output

jktang14 committed:
Add break statements

jktang14 committed:
Refactor Azir for OOP changes

jktang14 committed:
Handle empty list in hard disk

jktang14 committed:
Merge branch 'branch-A-MoreOOP'

jktang14 committed:
Move classes to Azir package

jktang14 committed:
Add package statement

jktang14 committed:
Merge branch 'A-Packages'

jktang14 committed:
Merge branch 'branch-A-Gradle'

jktang14 committed:
Edit main class configuration

jktang14 committed:
Remove exception handling

jktang14 committed:
Add exception handling

jktang14 committed:
Edit configuration for package

jktang14 committed:
Remove ParserTest

jktang14 committed:
Add additional todo check

jktang14 committed:
Add tests for parse method

jktang14 committed:
Fix string formatting

jktang14 committed:
Add tests for storage read method

jktang14 committed:
Merge branch 'branch-A-JUnit'

jktang14 committed:
Edit jar file naming

jktang14 committed:
Merge branch 'branch-A-Jar'

jktang14 committed:
Add javadoc for run method

jktang14 committed:
Add javaDoc comments

jktang14 committed:
Remove unused imports

jktang14 committed:
Update date formatting

jktang14 committed:
Add handling of find message

jktang14 committed:
Add logic for find message

jktang14 committed:
Add parsing for find command

jktang14 committed:
Merge branch 'branch-A-JavaDoc'

jktang14 committed:
Fix conflicts

jktang14 committed:
Merge branch-Level-9

hovanleong commented on src/main/java/Azir/Task.java:
```diff
@@ -0,0 +1,42 @@
+ * Representation of a task
+ * Able to set if a task has been completed
+ */
+public class Task {
```
Would you be open to considering making Task an abstract class, since you are likely to not have to instantiate it?

jktang14 replied:
Sure. That is a great idea.

hovanleong commented on src/main/java/Azir/Azir.java:
```diff
@@ -0,0 +1,149 @@
+                String fullCommand = ui.readCommand(obj);
+                ui.showLine();
+                String[] result = Parser.parse(fullCommand, tasks.getSize());
+                switch (result[0]) {
```
Great job on following the coding standards for switch statements!

hovanleong commented on src/main/java/Azir/Azir.java:
```diff
@@ -0,0 +1,149 @@
+package Azir;
```
Perhaps you could consider splitting your code into various sub-packages?

hovanleong commented on src/main/java/Azir/Azir.java:
```diff
@@ -0,0 +1,149 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Azir {
```
I think you could add JavaDoc comments for your Azir class when you have the time?

hovanleong reviewed (commented):
LGTM. Just some nits to fix. Overall really really great code and coding standards were also mostly obeyed. Keep up the great work!

Ma-Xizi commented on src/main/java/Azir/Azir.java:
```diff
@@ -0,0 +1,149 @@
+                    break;
+
+                case "find":
+                    ui.showCommandEndMessage(result[0]);
```
I feel the name 'showCommandEndMessage' is a bit not self-explanatory. Maybe 'replyListQuery' or other equivalent would be better?

jktang14 replied:
Good idea. Thanks

Ma-Xizi commented on src/main/java/Azir/Azir.java:
```diff
@@ -0,0 +1,149 @@
+                        String[] arr = description.split(" ");
+                        if (Arrays.stream(arr).anyMatch(str -> str.equals(result[1]))) {
+                            System.out.printf("%d. %s\n", i + 1, tasks.getTask(i));
+                        }
```
Good use of streams! 👍

Ma-Xizi commented on src/main/java/Azir/Azir.java:
```diff
@@ -0,0 +1,149 @@
+                    ui.showCommandEndMessage(result[0]);
+                    for (int i = 0; i < tasks.getSize(); i++) {
+                        String description = tasks.getTask(i).getDescription();
+                        String[] arr = description.split(" ");
```
```suggestion
                        String[] descriptionTokens = description.split(" ");
```

Ma-Xizi commented on src/main/java/Azir/Azir.java:
```diff
@@ -0,0 +1,149 @@
+            } catch (AzirException e) {
+                ui.showError(e.getMessage());
+            } finally {
+                ArrayList<String> lines = new ArrayList<>();
```
Could there be better names than lines?

Ma-Xizi commented on src/main/java/Azir/Ui.java:
```diff
@@ -0,0 +1,115 @@
+     * @param command User input
+     * @param task Corresponding task for command
+     */
+    public void showCommandEndMessage(String command, String task) {
```
Maybe can use more meaningful names such as replyTaskCommands?

Ma-Xizi reviewed (commented):
Overall good with slight modifications about naming suggested.

jktang14 committed:
Change comment naming

jktang14 committed:
Add user images

jktang14 committed:
Add javafx dependencies

jktang14 committed:
Edit main class

jktang14 committed:
Add fxml configurations

jktang14 committed:
Add dialog box

jktang14 committed:
Add Launcher

jktang14 committed:
Add main

jktang14 committed:
Add mainwindow

jktang14 committed:
Change methods to return strings

jktang14 committed:
Remove print statements

jktang14 committed:
Remove print statements

jktang14 committed:
Handle errors when loading hard disk

jktang14 committed:
Handle errors when loading hard disk

jktang14 committed:
Edit styling of fxmls

jktang14 committed:
Merge branch 'branch-Level-10'

jktang14 committed:
Enable assertions

jktang14 committed:
Add assert for switch statements

There are no checks for switch statements.

Assumptions are made that one of the cases definitely hits.

Let's add asserts in default statements to ensure that we are alerted
should we ever hit the default statement.

jktang14 committed:
Merge pull request #1 from branch-A-Assertions

Add assertions to files

jktang14 committed:
Add additional load message

jktang14 committed:
Remove unused methods

jktang14 committed:
Move task logic to helper function

jktang14 committed:
Remove excess import statement

jktang14 committed:
Merge pull request #2 from branch-A-CodeQuality

Edit code to follow code quality

jktang14 committed:
Merge branch 'master'

jktang14 committed:
Remove excess import statement

jktang14 committed:
Test PR workflow

jktang14 committed:
Merge branch 'master' into branch-A-Streams

jktang14 committed:
Merge pull request #3 from branch-A-Streams

Add stream functionality

jktang14 committed:
Remove test comment

jktang14 committed:
Add logic for duplicate handling

jktang14 committed:
Add method for checking duplicates

Duplicate tasks are allowed in the list.

Duplicate tasks would not make sense in a task list.

Let's add a checker for duplicates tasks.

jktang14 committed:
Merge branch 'branch-BCD-Extension'

jktang14 committed:
Edit speech of chatbot

jktang14 committed:
Add list of available commands

jktang14 committed:
Merge branch 'branch-A-Personality'

jktang14 committed:
Edit code to follow SLAP

jktang14 committed:
Remove bye case

jktang14 committed:
Edit code for SLAP

jktang14 committed:
Edit javadoc comments

jktang14 committed:
Edit javadoc comments

jktang14 committed:
Edit javadoc comments

jktang14 committed:
Edit javadoc comments

jktang14 committed:
Edit javadoc comments

jktang14 committed:
Remove irrelevant test

jktang14 committed:
Remove duke instances

jktang14 committed:
Add Ui screenshot

jktang14 committed:
Update README

jktang14 committed:
Add UI picture

jktang14 committed:
Add emoji

jktang14 committed:
Add jar link

jktang14 committed:
Add back bye functionality

jktang14 committed:
Add test
